### PR TITLE
DDF-2313: SourceConfigurationHandler Null pointer introduced

### PR DESCRIPTION
**Pull request by gordocanchola**
#### What does this PR
This PR adds null check to config.getProperties

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@clockard @ani6gup @mcalcote @vinamartin @ryeats 
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@figliold
@shaundmorris
#### How should this be tested?
Start up DDF with registry. 
Open the DDF Catalog app. 
Open the Node Information tab
Open the identity node and open the Services tab.
Add a service with at least one binding. Save
Open the Sources tab.
Verify a source exists for you identity node, you may have to refresh the list.
#### What are the relevant tickets?
DDF-2313
#### Checklist:
- [ ] Documentation Updated
- [X] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

Fixing Null Pointer exception.

### CODE REVIEW

File: catalog/spatial/registry/registry-source-configuration-handler/src/main/java/org/codice/ddf/registry/source/configuration/SourceConfigurationHandler.java [hunk 1]
```diff
@@ -570,6 +577,20 @@ private String getBindingTypeToActivate(List<ServiceBindingType> bindingTypes) {
         return properties;
     }
 
+    private String getConfigStringProperty(Configuration config, String property) {
+        if (config.getProperties() == null) {
```
Comment (shaundmorris) [inline catalog/spatial/registry/registry-source-configuration-handler/src/main/java/org/codice/ddf/registry/source/configuration/SourceConfigurationHandler.java]: Should you go ahead and validate both parameters for null? So check `config` and `property`?

File: catalog/spatial/registry/registry-source-configuration-handler/src/main/java/org/codice/ddf/registry/source/configuration/SourceConfigurationHandler.java [hunk 2]
```diff
@@ -570,6 +577,20 @@ private String getBindingTypeToActivate(List<ServiceBindingType> bindingTypes) {
...
+            return "";
+        }
+
```
Comment (figliold) [inline catalog/spatial/registry/registry-source-configuration-handler/src/main/java/org/codice/ddf/registry/source/configuration/SourceConfigurationHandler.java]: Is it always safe to cast to String?

Comment (gordocanchola) [inline catalog/spatial/registry/registry-source-configuration-handler/src/main/java/org/codice/ddf/registry/source/configuration/SourceConfigurationHandler.java]: Added instanceof check.

File: catalog/spatial/registry/registry-source-configuration-handler/src/main/java/org/codice/ddf/registry/source/configuration/SourceConfigurationHandler.java (hunk 1 above)
Comment (gordocanchola) [inline catalog/spatial/registry/registry-source-configuration-handler/src/main/java/org/codice/ddf/registry/source/configuration/SourceConfigurationHandler.java]: Added checks for both.

**Comment by vinamartin** [comment]
Lgtm 👍

**Comment by shaundmorris** [comment]
👍

**Comment by clockard** [comment]
Build and manual tests passed 👍

**Pull request merged**